pipeline/authn/authenticator_bearer_token.go

```diff
@@ -0,0 +1,109 @@
+package authn
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/pkg/errors"
+	"github.com/tidwall/gjson"
+
+	"github.com/ory/go-convenience/stringsx"
+
+	"github.com/ory/oathkeeper/driver/configuration"
+	"github.com/ory/oathkeeper/helper"
+	"github.com/ory/oathkeeper/pipeline"
+)
+
+func init() {
+	gjson.AddModifier("this", func(json, arg string) string {
+		return json
+	})
+}
+
+type AuthenticatorBearerTokenFilter struct {
+}
+
+type AuthenticatorBearerTokenConfiguration struct {
+	CheckSessionURL     string                      `json:"check_session_url"`
+	BearerTokenLocation *helper.BearerTokenLocation `json:"token_from"`
+	PreservePath        bool                        `json:"preserve_path"`
+	ExtraFrom           string                      `json:"extra_from"`
+	SubjectFrom         string                      `json:"subject_from"`
+}
+
+type AuthenticatorBearerToken struct {
+	c configuration.Provider
+}
+
+func NewAuthenticatorBearerToken(c configuration.Provider) *AuthenticatorBearerToken {
+	return &AuthenticatorBearerToken{
+		c: c,
+	}
+}
+
+func (a *AuthenticatorBearerToken) GetID() string {
+	return "bearer_token"
+}
+
+func (a *AuthenticatorBearerToken) Validate(config json.RawMessage) error {
+	if !a.c.AuthenticatorIsEnabled(a.GetID()) {
+		return NewErrAuthenticatorNotEnabled(a)
+	}
+
+	_, err := a.Config(config)
+	return err
+}
+
+func (a *AuthenticatorBearerToken) Config(config json.RawMessage) (*AuthenticatorBearerTokenConfiguration, error) {
+	var c AuthenticatorBearerTokenConfiguration
+	if err := a.c.AuthenticatorConfig(a.GetID(), config, &c); err != nil {
+		return nil, NewErrAuthenticatorMisconfigured(a, err)
+	}
+
+	if len(c.ExtraFrom) == 0 {
+		c.ExtraFrom = "extra"
+	}
+
+	if len(c.SubjectFrom) == 0 {
+		c.SubjectFrom = "subject"
```

Comment:
OAuth2 uses the shorthand `sub` - I think we should use `sub` per default also:

```suggestion
		c.SubjectFrom = "sub"
```